### ISSUE TITLE
the round-robin do not take effective version 0.5.9

### ISSUE THREAD

**Issue by leland17** [etcd-io/jetcd#1033]
**Describe the bug**
when a node is down in my etcd cluster, due to client pick_first mechanism,  the client would not connected to the cluster. even retry action do not taken. 

When i change it to the round_robin policy, do not take efffective too.

**To Reproduce**
1. have a three node cluster, for example, 10.xx.xx.1, 10.xx.xx.2, 10.xx.xx.3, and the 10.xx.xx.1 is down, then the configure sequence as follows: 10.xx.xx.1, 10.xx.xx.2, 10.xx.xx.3,  the cluster do not connectd. But change the sequence to 10.xx.xx.2, 10.xx.xx.2, 10.xx.xx.3, then it could connect to the cluster.

**Expected behavior**
the sdk client should retry the good node to connect to the cluster

**Additional context**

**Comment by lburgazzoli** [comment]
What versions did you use ?

**Comment by leland17** [comment]
> What versions did you use ?

client 0.5.9

**Comment by leland17** [comment]
> What versions did you use ?

server 3.4.14.3

**Comment by lburgazzoli** [comment]
> > What versions did you use ?
> 
> client 0.5.9

An issue with the way the IP resolver works has been fixed in 0.6.x, can you try with that one ?

**Comment by leland17** [comment]
where can i download that jar？ 
![image](https://user-images.githubusercontent.com/9267662/146173866-185b3046-ea56-461d-8aaf-25812bca8b4d.png)

**Comment by lburgazzoli** [comment]
From maven https://search.maven.org/search?q=jetcd-core

**Comment by leland17** [comment]
i found that but it seems do not have jetcd-all 0.6.1

**Comment by lburgazzoli** [comment]
It is expected, jetcd-all has been removed in 0.6.x
If you use maven/gradle, the right set transitive dependencies should be retrieved by jetcd-core

**Comment by leland17** [comment]
unfortunately, it has been the same.
nothing helped. and have no idea now, need help

**Comment by lburgazzoli** [comment]
does etcdctl works in such set-up ?
jetcd does nothing more than delegating to grpc-java

**Comment by leland17** [comment]
I don't know why the retryenable is false, and i have no ability to change it

![image](https://user-images.githubusercontent.com/9267662/146190684-825d62f6-87d8-4950-9dc9-f80a47e3d5a3.png)

**Comment by lburgazzoli** [comment]
I mean, you need to try with etcdctl with a cluster and killing one node to wee how it works.

**Comment by leland17** [comment]
> I mean, you need to try with etcdctl with a cluster and killing one node to wee how it works.

it works of course the whole cluster. Because my go-clientV3 runs as normal

**Comment by lburgazzoli** [comment]
can you try to debug it ? 
the retry logic should be implemented by the application code so maybe, there is some exception that is not taken into account

**Comment by leland17** [comment]
![image](https://user-images.githubusercontent.com/9267662/146203642-032d4ac0-bc59-44e1-9e7b-6efe9ef2b0e3.png)
how should i set authority if my endpoints is in picture. and my cert do not contains such ip list

**Comment by leland17** [comment]
> [there is an authority option in the builder](https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/ClientBuilder.java#L299-L315)

thanks a lot, problem have been solved.

**Comment by leland17** [comment]
> [there is an authority option in the builder](https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/ClientBuilder.java#L299-L315)

but also I want to know，whether  the authority supports multi ip:port config, like: 10.xx.xx.1:2379, 10.xx.xx.2:2379, 10.xx.xx.3:2379.  and the code comment  never details

**Comment by lburgazzoli** [comment]
> > [there is an authority option in the builder](https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/ClientBuilder.java?rgh-link-date=2021-12-16T01%3A31%3A45Z#L299-L315)
> 
> thanks a lot, problem have been solved.
> 
> but the jetcd client do not have any debug/warn log. due to the ssl shakehands (ip not match) failed, the thread is kept trying to connect and not connected.

if you have time, please send a PR to add more visibility

**Comment by lburgazzoli** [comment]
> > [there is an authority option in the builder](https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/ClientBuilder.java?rgh-link-date=2021-12-16T08%3A45%3A35Z#L299-L315)
> 
> but also I want to know，whether the authority supports multi ip:port config, like: 10.xx.xx.1:2379, 10.xx.xx.2:2379, 10.xx.xx.3:2379. and the code comment never details

This code is what configures the authority: https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/main/java/io/etcd/jetcd/resolver/IPNameResolver.java


So in essence, if the authority is configured with the builder, then it is used otherwise it is derived from the addresses as this is what grpc-java offers. See the NameResolver class

**Comment by leland17** [comment]
And whether by the pick_first policy (the first node is ther error node),  jetcd have a buffer to storage context deadline error write request. And after the node recover，then the buffer data are transfered to write into the etcd cluster.

In my case that take place.

**Comment by lburgazzoli** [comment]
can you rephrase the sentence ?

**Comment by leland17** [comment]
> can you rephrase the sentence ?

I hava edit my question. sorry

**Comment by lburgazzoli** [comment]
> does jetcd have a mechanism to storage the failed put request (context deadline error) ? and after the abnormal node is recover, the failed request will retry to send to the cluster ? the loadbalance policy is pick_first

jetcd as a retry logic but the etcd node to hit is delegated to grpc-java (by the load balance policy)
